lib/compile/csdl2openapi.js

```diff
@@ -453,14 +453,25 @@ module.exports.csdl2openapi = function (
      * @param {object} serversObject Input servers object
      * @return {Array} The list of servers
      */
-    function getServers(serviceRoot, serversObject) {
+    function getServers(serviceRoot, serversObject, csdl) {
         let servers;
 
         if (serversObject) {
             try {
-                servers = JSON.parse(serversObject);
+              servers = JSON.parse(serversObject);
+              // if csdl.$Version is 4.01 then protocol is rest if it is less than 4.01 then protocol is odata
+              const protocol = csdl.$Version <= "4.01" ? "odata/v4" : "rest";
+              // append /protocol/{$serviceName} to the URL
+              let serviceName = csdl.$EntityContainer.includes(
```

Comment:
Can we recheck this? Editing `csdl.$EntityContainer` is not ideal. I'm also not sure why we are checking for `.EntityContainer`.

Reply:
we have to get the service name which is stored here

Reply:
yes, but how it is done is not right

Reply:
made changes, now we take the service name like how we do for other funtions